Update the Price Chart feature to remove the ajax call (to do the call from the server)

mariarabin opened the pull request:
1) Updated the priceChart.js file to remove the ajax code and create the plotChart function to return only the (dataSets, dataSetsLabel)
2) Updated the data.js to add the plotChart function under the updatingChart main function
3) Added test code - test data for apiRoutes.js file
4) Updated the dataFetch.js to add a dataFetchManager.getQuotes function
5) Updated the iexinterface.js to include the iexRequest.test.intraday & iexRequest.test.historicalPrices functions
6) Updated the home.handlebars to replace the function 'processRequest' for an onclick event to now be 'UpdatingChart' function

mariarabin committed:
1) Updated the priceChart.js file to remove the ajax code and create the plotChart function to return only the (dataSets, dataSetsLabel)
2) Updated the data.js to add the plotChart function under the updatingChart main function
3) Added test code - test data for apiRoutes.js file
4) Updated the dataFetch.js to add a dataFetchManager.getQuotes function
5) Updated the iexinterface.js to include the iexRequest.test.intraday & iexRequest.test.historicalPrices functions
6) Updated the home.handlebars to replace the function 'processRequest' for an onclick event to now be 'UpdatingChart' function

mariarabin committed:
Merge branch 'master' into priceChart-Server

mariarabin committed:
1) Updated the home.handlers file to have the priceChartDisplay id

alexjzyang commented on routes/apiRoutes.js:
```diff
@@ -66,6 +66,11 @@ module.exports = function (app) {
+        result = {
+            dataSets =[],
+            dataSetsLabel =[]
+        }
```
These lines are what we wrote as examples. these new lines shouldn't be there.

mariarabin replied:
Removed already.

alexjzyang commented on public/js/data.js:
```diff
@@ -70,9 +70,11 @@ async function updateValue(metricId) {
 
-function updatingChart(period) {
+//updating the Price Chart display
+function updatingChart(period, chartType, btnElemID) {
```
I believe this function only need to take in period as a single argument. If this is the case could you please delete chartType, and btnElemID?
Do the same in the html handlebar template please

mariarabin replied:
Done.

alexjzyang commented on views/home.handlebars:
```diff
@@ -1,186 +1,178 @@
+                                    onClick="updatingChart('1y', '/chart/', 'OneYearID')"
+                                    class="badge badge-secondary">1-Year</button>
+                                <button type="button" id="AllID" onClick="updatingChart('max', '/chart/', 'AllID')"
+                                    class="badge badge-secondary">ALL</button>
```
These lines above, the updatingChart should only need period. then we parse period in the backend. for intra-day, best to pass something like "1d".
Let's do "5y" instead of "max" because I think max plots are always very slow.

mariarabin replied:
Done

alexjzyang reviewed (changes requested):
Hi @mariarabin, Great job! I like what you've done with the dataFetchManager and the IEX Request object. 
Don't forget to pull and merge master. I changed slightly the implementation of the IEX request object to take into account of the production enable environment variable.

mariarabin committed:
Updated based on the comments made
1) data.js - argument to be only the 'Period' under the updatingChart function
2) apiRoutes.js - removed the TEST CODE
3) home.handlebars - argument to be only the 'Period' under the updatingChart function and '' for intraday is now '1d'

alexjzyang reviewed (approved):
Thanks @mariarabin
